hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcClient.java

```diff
@@ -599,10 +599,8 @@ public Object invokeMethod(
           }
           LOG.error("Cannot get available namenode for {} {} error: {}",
               nsId, rpcAddress, ioe.getMessage());
-          if (this.namenodeResolver != null) {
-            this.namenodeResolver.rotateCache(nsId, namenode, shouldUseObserver);
-            LOG.info("Rotate cache of pair: <ns: {}, observer use: {}>", nsId, shouldUseObserver);
-          }
+          // Rotate cache so that client can retry the next namenode in the cache
+          this.namenodeResolver.rotateCache(nsId, namenode, shouldUseObserver);
```

Comment:
There is no need to judge whether the namenodeResolver is empty, because the namenodeResolver is not null when the router is initialized
in Router.java line:178

``` java
   this.namenodeResolver = newActiveNamenodeResolver(
       this.conf, this.stateStore);
   if (this.namenodeResolver == null) {
     throw new IOException("Cannot find namenode resolver.");
   }
```